### ISSUE TITLE
combine()?

### ISSUE THREAD

**Issue by possibly** [tillarnold/fixed-2d-array#5]
So I am making this game (same one as before) and am still using Fixed2DArray. I would like to have my Fixed2DArray grow as the player 'explores' it, kinda like how minecraft generates new terrain as the player travels around the world. Arrays in Javascript have [concat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat), but i'd like Fixed2DArray to have a concat that works a bit differently. Im thinking about calling Fixed2DArray's concat()-equivalent combine().

For example:

```
var fixedArray = require('..');
var fa1 = new fixedArray(2,2,0);
var fa2 = new fixedArray(2,2,1);
var fa3 = fa1.combine(fa2);
console.log(fa3._grid); 
```

Output

```
[
[0, 0, 1, 1],
[0, 0, 1, 1]
]
```

HOWEVER, what if I wanted to combine `fa2` to the top of `fa1`, as opposed to the right? Like so:

```
[
[1, 1],
[1, 1],
[0, 0],
[0, 0],
]
```

wowowowow.

Then using combine would like more like

```
var fa3 = fa1.combine(fa2, 'top'); 
//left, right, top, bottom
```

But what about fixedArrays of a different sizes? For example,

```
var fixedArray = require('..');
var fa4 = new fixedArray(2,2,4);
var smallerFA5 = new fixedArray(1,1,5);
var fa6 = fa4.combine(smallerFA5);
console.log(fa6._grid); 
```

What should `fa6` be? Outputting a Fixed2DArray might be strange since the combination of a 2x2 Fixed2DArray and a 1x1 Fixed2DArray leads to a "Staggered2DArray", not a fixed2DArray!

Example of "Staggered2DArray":

```
[
[4, 4, 5]
[4, 4]
]
```

I have a few ideas, though I am not sure which one is best:

1) Fixed2DArray's functions still work if the `._grid` variable is technically "staggered." Therefore, the output of `fa6` can just be a Fixed2DArray that has had its `._grid` variable played with.

2) Only allow `combine()` to "combine" Fixed2DArrays of the same height and width. I think this is how [typical mathematics matrices](https://www.npmjs.com/package/matrixmath) work?

3) Keep the `._grid` "fixed" but use some filler variable. For example, the output of `fa6` could be:

```
[
[4, 4, 5],
[4, 4, null]
]
```

if the filler variable was null. `combine()` could have an optional `fillerVar` argument as well, so the user can define whatever `fillerVar` works for them.

I dont like option 1. Option 2 is easiest. Option 3 is kinda cool, but I would like to notify the coders in some way that "hey, these Fixed2DArrays are different if you werent aware!"

**Comment by tillarnold** [comment]
Yes I don't like option 1 as well. Option 3 with an optional fillerVar argument sounds like the most flexible option. Option 2 sounds ok as well.

**Comment by possibly** [comment]
After re-reading this, I think I agree with you. Option 2 sounds the most intuitive for an object called Fixed2DArray.
